query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+            + "order by start_time_millis desc , end_time_millis limit 1000, 100",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithGroupByWithMultipleAggregates() {
+    QueryRequest orderByQueryRequest = buildMultipleGroupByMultipleAggQuery();
+    Builder builder = QueryRequest.newBuilder(orderByQueryRequest);
+    builder.setLimit(20);
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "select service_name, span_name, count(*), avg(duration_millis) FROM public.\"span-event-view\""
+            + " where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " group by service_name, span_name limit 20",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithGroupByWithMultipleAggregatesAndOrderBy() {
+    QueryRequest orderByQueryRequest = buildMultipleGroupByMultipleAggAndOrderByQuery();
+    Builder builder = QueryRequest.newBuilder(orderByQueryRequest);
+    builder.setLimit(20);
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "select service_name, span_name, count(*), avg(duration_millis) FROM public.\"span-event-view\""
+            + " where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " group by service_name, span_name order by service_name, avg(duration_millis) desc , count(*) desc  limit 20",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithDistinctCountAggregation() {
+    Filter startTimeFilter =
+        createTimeFilter("Span.start_time_millis", Operator.GT, 1570658506605L);
+    Filter endTimeFilter = createTimeFilter("Span.end_time_millis", Operator.LT, 1570744906673L);
+    QueryRequest queryRequest =
+        QueryRequest.newBuilder()
+            .addAggregation(
+                createAliasedFunctionExpression(
+                    "DISTINCTCOUNT", "Span.id", "distinctcount_span_id"))
+            .setFilter(
+                Filter.newBuilder()
+                    .setOperator(Operator.AND)
+                    .addChildFilter(startTimeFilter)
+                    .addChildFilter(endTimeFilter)
+                    .build())
+            .setLimit(15)
+            .build();
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "select count(distinct encode(span_id, 'hex')) FROM public.\"span-event-view\""
```

Comment:
we are missing alias `distinctcount_span_id` in the final query, would that be an issue?

Reply:
Will handle alias related changes in separate PR